src/irc.lua

```diff
@@ -6,9 +6,13 @@ local ssl = require 'ssl'
 
 irc.init = function (irc_config)
     local bare = socket.connect(irc_config.address, irc_config.port)
-    local conn = ssl.wrap(bare, irc_config.sslparams)
-    conn:dohandshake()
-    return conn
+    if irc_config.use_ssl then
+        local conn = ssl.wrap(bare, irc_config.sslparams)
+        conn:dohandshake()
+        return conn
+    else
+        return bare
```

Comment:
Can this also print a diagnostic message (something along the lines of `WARNING: you have not configured TLS; your connection will be insecure`)?